src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.ts

```diff
@@ -61,11 +59,19 @@ export class RgwMultisiteZoneFormComponent implements OnInit {
     public rgwZoneGroupService: RgwZonegroupService,
     public notificationService: NotificationService,
     public rgwUserService: RgwUserService,
-    public modalService: ModalService
+    public modalService: ModalCdsService,
+    @Optional() @Inject('resource') public resource: string,
+    @Optional() @Inject('defaultsInfo') public defaultsInfo: string[],
+    @Optional() @Inject('multisiteInfo') public multisiteInfo: object[],
+    @Optional() @Inject('action') public action: string,
+    @Optional() @Inject('info') public info: any,
+    @Optional() @Inject('editing') public editing: boolean
   ) {
-    this.action = this.editing
-      ? this.actionLabels.EDIT + this.resource
-      : this.actionLabels.CREATE + this.resource;
+    super();
+    console.log(this.action, 'action');
+    this.action =
+      this.action === this.actionLabels.EDIT ? this.actionLabels.EDIT : this.actionLabels.CREATE;
```

Comment:
you can remove this right. Its redundant if you pass it correctly from the child component.